src/extensions/CanvasWidget/CanvasWidget.tsx

```diff
@@ -77,14 +77,15 @@ const dragReducer = (state: DragState, action: DragAction): DragState => {
     throw Error('Unexpected: unreachable, but keeps ESLint happy')
 }
 
-interface Props<T extends BaseLayerProps> {
-    layers: CanvasWidgetLayer<T, any>[] | null, // the layers to paint (each corresponds to a canvas html element)
+interface Props {
+    layers: (CanvasWidgetLayer<any, any> | null)[], // the layers to paint (each corresponds to a canvas html element)
+    width: number,
+    height: number
     preventDefaultWheel?: boolean // whether to prevent default behavior of mouse wheel
-    layerProps: T // props sent to the layer
 }
 
-const CanvasWidget = <T extends BaseLayerProps>(props: Props<T>) => {
-    const { layers, layerProps, preventDefaultWheel } = props
+const CanvasWidget = (props: Props) => {
```

Comment:
CanvasWidget no longer has the layer props as a generic, because the layers can have different props.